apphub/image_generation/pggan_nihchestxray/pggan_nihchestxray.py

```diff
@@ -154,62 +155,79 @@ def on_epoch_end(self, state):
                 else:
                     plt.imshow(disp_img)
                 plt.axis('off')
-            plt.savefig(os.path.join(self.save_dir, 'image_at_{:08d}.png').format(state["epoch"]))
+                disp_img = np.uint8(disp_img * 255)
+                cv2.imwrite(os.path.join(self.save_dir, 'image_at_{:08d}_{}.png').format(state["epoch"], i), disp_img)
+            plt.savefig(os.path.join(self.save_dir, 'image_grid_{:08d}.png').format(state["epoch"]))
             print("on epoch {}, saving image to {}".format(state["epoch"], self.save_dir))
 
 
+class ResetOptimizer(Trace):
+    def __init__(self, reset_epochs, optimizer):
+        super().__init__(inputs=None, outputs=None, mode="train")
+        self.reset_epochs = reset_epochs
+        self.optimizer = optimizer
+
+    def on_epoch_begin(self, state):
+        if state["epoch"] in self.reset_epochs:
+            for weight in self.optimizer.weights:
+                backend.set_value(weight, weight - weight)
```

Comment:
weight - weight instead of zero?

Reply:
`weight - weight` preserves the shape and data type of different weight.    otherwise we will have to querry the shape and type.

Reply:
ah, clever